api/src/pcapi/core/users/api.py

```diff
@@ -1316,3 +1316,54 @@ def delete_old_trusted_devices() -> None:
 
     users_models.TrustedDevice.query.filter(users_models.TrustedDevice.dateCreated <= five_years_ago).delete()
     db.session.commit()
+
+
+def _get_users_with_suspended_account_to_notify(expiration_delta_in_days: int) -> list[models.User]:
+    start = datetime.date.today() - datetime.timedelta(days=expiration_delta_in_days)
+
+    # distinct keeps the first row if duplicates are found. Since rows
+    # are ordered by userId and eventDate, this query will fetch the
+    # latest event for each userId.
+    user_ids_and_latest_action = (
+        users_models.User.query.distinct(history_models.ActionHistory.userId)
+        .join(users_models.User.action_history)
+        .filter(
+            sa.or_(
+                history_models.ActionHistory.actionType == history_models.ActionType.USER_SUSPENDED,
+                history_models.ActionHistory.actionType == history_models.ActionType.USER_UNSUSPENDED,
+            )
+        )
+        .order_by(history_models.ActionHistory.userId, history_models.ActionHistory.actionDate.desc())
+        .with_entities(
+            users_models.User.id,
+            history_models.ActionHistory.actionDate,
+            history_models.ActionHistory.actionType,
+            history_models.ActionHistory.extraData["reason"].astext.label("reason"),
+        )
+    ).subquery()
+
+    # Deletion is a special case of suspension, no need to delete again
+    # an already deleted account.
+    query = users_models.User.query.join(
+        user_ids_and_latest_action, users_models.User.id == user_ids_and_latest_action.c.id
+    ).filter(
+        user_ids_and_latest_action.c.actionDate - start >= datetime.timedelta(days=0),
```

Comment:
these(1350-1351) are the two lines that are different between this function and https://github.com/pass-culture/pass-culture-main/blob/b6f5adbaf43b506ab7bfe9416b277b480b3da76e/api/src/pcapi/core/fraud/api.py#L619

Reply:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** ‏S'il y a que 2 lignes de différence, on doit pouvoir éviter autant de duplication
Par exemple avec des fonctions communes (une qui construit la query par exemple)

Reply:
je peux rajouter un paramettre exact:bool,  si c'est true je verifie que c'est le même jour comme ici sinon je prend tout ce qui vient après.
t'en pense quoi?
sinon je ne vois pas trop comment faire en laissant les choses propres.
si t'as envie on peux en discuter à vive voix